File: libgdx_drag_race/core/src/com/nopalsoft/dragracer/objects/SpeedBar.java
package com.nopalsoft.dragracer.objects;

import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.scenes.scene2d.Actor;
import com.nopalsoft.dragracer.Assets;

public class SpeedBar extends Actor {

    public float maxLife;
    public float actualLife;

    public SpeedBar(float maxLife, float x, float y, float width, float height) {
        this.setBounds(x, y, width, height);
        this.maxLife = maxLife;
        this.actualLife = maxLife;
    }

    public void updateActualLife(float actualLife) {
        if (actualLife >= maxLife) {
            actualLife = maxLife;
        }
        this.actualLife = actualLife;
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {

        batch.draw(Assets.redMarkerBar, this.getX(), this.getY(),
                this.getWidth(), this.getHeight());
        if (actualLife > 0)
            batch.draw(Assets.greenMarkerBar, this.getX(), this.getY(),
                    this.getWidth() * (actualLife / maxLife), this.getHeight());
    }
}

File: The_True_Color/core/src/com/nopalsoft/dragracer/objects/SpeedBar.kt
package com.nopalsoft.dragracer.objects

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.scenes.scene2d.Actor
import com.nopalsoft.dragracer.Assets

class SpeedBar(maxLife: Float, x: Float, y: Float, width: Float, height: Float) : Actor() {
    private var maxLife: Float
    private var actualLife: Float

    init {
        this.setBounds(x, y, width, height)
        this.maxLife = maxLife
        this.actualLife = maxLife
    }

    fun updateLifeBar(newLifeLevel: Float) {
        var actualLife = newLifeLevel
        if (actualLife >= maxLife) {
            actualLife = maxLife
        }
        this.actualLife = actualLife
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        batch.draw(
            Assets.redMarkerBar!!, this.x, this.y,
            this.width, this.height
        )
        if (actualLife > 0) batch.draw(
            Assets.greenMarkerBar!!, this.x, this.y,
            this.width * (actualLife / maxLife), this.height
        )
    }
}
